File: auth-service/src/main/java/com/pyonpyontech/authservice/model/customer/Outlet.java
package com.pyonpyontech.authservice.model.customer;

import com.pyonpyontech.authservice.model.customer_service_report.CsrReport;
import com.pyonpyontech.authservice.model.pest_control.Employee.Supervisor;
import com.pyonpyontech.authservice.model.pest_control.Employee.Technician;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table
public class Outlet {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "customer_id")
    private Customer customer;

    @Column(name = "region", nullable = false)
    private String region;

    @Column(name = "address", nullable = false)
    private String address;

    @ManyToOne
    @JoinColumn(name="supervisor_id", nullable=false)
    private Supervisor supervisor;

    @ManyToOne
    @JoinColumn(name="technician_id")
    private Technician technician;

    @OneToMany(mappedBy = "outlet")
    private List<CsrReport> reports;
}

File: frontend-service/utils/models/customer/customer/outlet.ts
export interface Outlet {
    id: number;
    name: string;
    region: string;
    address: string;
    supervisor: { id: number, user: { name: string }};
  }
  
export class OutletClass implements Outlet {
  id: number;
  name: string;
  region: string;
  address: string;
  supervisor: { id: number, user: { name: string }};

  constructor(obj: any) {
    this.id = obj.id;
    this.name = obj.name;
    this.region = obj.region;
    this.address = obj.address;
    this.supervisor = obj.supervisor;
  }
} 